### PR TITLE
HEEDLS-900 admin self-registration fixes

### PR DESCRIPTION
**Pull request by livzorn**
### JIRA link
[_HEEDLS-900_](https://softwiretech.atlassian.net/browse/HEEDLS-900)

### Description
Removed the word "address" from what the screen reader reads out for changing an email.
Added Terms Agreed to the columns updated when registering a new user.

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [ ] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [ ] Updated/added documentation in Swiki and/or Readme. Links (if any) are below:
  - [doc_1_here](link_1_here)
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

File: DigitalLearningSolutions.Data/DataServices/RegistrationDataService.cs [hunk 1]
```diff
@@ -47,6 +47,7 @@ public string RegisterNewUserAndDelegateAccount(DelegateRegistrationModel delega
                 delegateRegistrationModel.Active,
                 PasswordHash = "temp",
                 ProfessionalRegistrationNumber = (string?)null,
+                TermsAgreed = currentTime,
```
Comment (DanBloxham-sw) [inline DigitalLearningSolutions.Data/DataServices/RegistrationDataService.cs:50]: Couple of things on this.
Firstly, I'm not sure if we do want to set TermsAgreed. Looking at the old code, and the old ticket, we never set TC_Agreed which I think was the old version of this field (used to be on AdminUsers). @SteveJacksonSoft do you know if the requirement to set this has slipped off the radar somehow?

Secondly, this method is currently used in other delegate registration process. So I think we want to be careful about setting this here. I presume a non-null date basically means the user has checked the "I agree to the Terms and Conditions..." check box when registering on said date. But it looks like this method is used in the DelegateBulkUpload journey, where the user certainly wouldn't have checked that check box themselves.

**Review by DanBloxham-sw — COMMENTED** [review]
Looks like this will solve the issues, but I am unsure on the TermsAgreed portion.

File: DigitalLearningSolutions.Data/DataServices/RegistrationDataService.cs (hunk 1 above)
Comment (simonbrent) [reply DigitalLearningSolutions.Data/DataServices/RegistrationDataService.cs:50]: I agree...I think we'd need to pass the state of "has the user being registered checked a Ts&Cs checkbox during this journey" through the code until we reach this point in order to set `TermsAgreed` in the database

Comment (SteveJacksonSoft) [reply DigitalLearningSolutions.Data/DataServices/RegistrationDataService.cs:50]: Yeah, we should be populating TC_Agreed in admin registration (external and internal). Don't know how that got missed. And we should be passing the checkbox value through in order to set it, as Simon says.

**Review by simonbrent — CHANGES_REQUESTED** [review]
It would be good to:
* Update the `RegisterNewUserAndDelegateAccount_sets_all_fields_correctly_on_registration` test so that `registerJourneyContainsTermsAndConditions` is true, and the test checks that `TermsAgreed` is set to a reasonable timestamp
* A new test is added to check that `TermsAgreed` is null if `registerJourneyContainsTermsAndConditions` is false

File: DigitalLearningSolutions.Data/DataServices/RegistrationDataService.cs (hunk 1 above)
Comment (SteveJacksonSoft) [reply DigitalLearningSolutions.Data/DataServices/RegistrationDataService.cs]: I've raised [HEEDLS-964](https://softwiretech.atlassian.net/browse/HEEDLS-964) to fix this for the admin registration journey on master (we need to include it in the upcoming release).

**Review by simonbrent — APPROVED** [review]

**Pull request merged**